config/confighttp/confighttp.go

```diff
@@ -262,6 +263,14 @@ func WithErrorHandler(e errorHandler) ToServerOption {
 	}
 }
 
+// WithDecoder provides support for additional decoders to be configured
+// by the caller.
+func WithDecoder(key string, dec decoder) ToServerOption {
```

Comment:
See https://github.com/open-telemetry/opentelemetry-collector/pull/7870 related to exposing internal types

Reply:
Yup, i was wondering the same thing when i looked at errorHandler :D

Reply:
Removed internal type